card/mod/05_email/set/all/follow.rb

```diff
@@ -2,30 +2,33 @@
 
 FOLLOWER_IDS_CACHE_KEY = 'FOLLOWER_IDS'
 
-event :cache_expired_for_new_set, before: :store, on: :create,
-  when: proc { |c| c.type_id == Card::SetID } do
+# FIXME: this should be in type/set
+event :cache_expired_for_new_set,
+      before: :store, on: :create,
+      when: proc { |c| c.type_id == Card::SetID } do
   Card.follow_caches_expired
 end
 
-event :cache_expired_for_type_change, before: :store, changed: :type_id do
-  #FIXME expire (also?) after save
+event :cache_expired_for_type_change, before: :store, on: :update,
+                                      changed: :type_id do
```

Comment:
the `on: :update` here and in the next event make a big difference.  `changed: :type_id` doesn't exclude all create events (where the type_id is changed from nothing to something, I suppose), so the rule cache was getting cleared on ALL create events.